src/NServiceBus.Core/Recoverability/Faults/FaultContext.cs

```diff
@@ -6,15 +6,25 @@
 
     class FaultContext : BehaviorContext, IFaultContext
     {
-        public FaultContext(OutgoingMessage message, string errorQueueAddress, Exception exception, IBehaviorContext parent)
+        public FaultContext(OutgoingMessage message, string localAddress, Exception exception, IBehaviorContext parent)
             : base(parent)
         {
             Guard.AgainstNull(nameof(message), message);
-            Guard.AgainstNullAndEmpty(nameof(errorQueueAddress), errorQueueAddress);
+            Guard.AgainstNullAndEmpty(nameof(localAddress), localAddress);
             Guard.AgainstNull(nameof(exception), exception);
+
+            var errorQueueAddress = ErrorQueueSettings.GetConfiguredErrorQueue(Builder.Build<Settings.ReadOnlySettings>());
```

Comment:
Would you put this directly into the context or into a behavior that targets this context? That way the behavior can be initialized with the error Q address in the feature that sets up the fault context.